docs/proposals/005_carbon_loading_mechanism.md

```diff
@@ -0,0 +1,252 @@
+
+**Diagramm**
+
+```bash
+                                    +--------------------+
+                                    |  User (Browser)    |
+                                    +--------------------+
+                                             |
+                                             v
+                                    +---------------------+
+                                    | Shell App (Ingress) |
+                                    | Service: shell-app  |
+                                    +---------------------+
+                                             |
+                          +------------------+------------------+
+                          |                                     |
+                          v                                     v
+           +--------------------------+          +--------------------------+
+           |    Extension XYZ (Pod)   |          |    Extension ABC (Pod)   |
+           | Service: extension-xyz   |          | Service: extension-abc   |
+           | Exposes: /bundle.js      |          | Exposes: /bundle.js      |
+           | Exposes: /entrypoint.js  |          | Exposes: /entrypoint.js  |
+           +--------------------------+          +--------------------------+
+                          |                                     |
+                          v                                     v
+           +--------------------------+          +--------------------------+
+           | Deployment: extension-xyz|          | Deployment: extension-abc|
+           | Image: xyz:latest        |          | Image: abc:latest        |
+           +--------------------------+          +--------------------------+
+```
+
+### Key Concepts
+
+- Extensions are hosted externally and loaded dynamically at runtime.
+- A manifest file lists the extensions and their versions available for loading.
+- Internal and external extensions are bundled and deployed to a designated CDN.
+- External extensions are bundled and deployed to a predefined CDN.
+- The shell application references the CDN URL in the manifest file for seamless integration and dynamic loading at runtime.
+- Kubernetes setup for deploying a shell app and multiple extensions as Docker images, with seamless internal communication.
+
+### Pros and Cons
+
+- **Pros**:
+
+  - **Lightweight Shell App**: Keeps the shell application lightweight by utilizing resources hosted externally.
+  - **Centralized Management**: A single manifest file simplifies the management of extensions by providing centralized information for runtime loading.
+  - **Fault Isolation**: Faulty extensions do not affect the deployment of others, allowing independent rollbacks.
+  - **Dogfooding**: Utilizing your own Docker images for extensions deployed in Kubernetes not only demonstrates confidence in your ecosystem but also validates and tests the artifacts hosted in GHCR, ensuring reliability and alignment with your offered solutions.
```

Comment:
- Smaller size of the bundle and ability to deploy shell app and extensions independently

Reply:
Is this not covered in line 232? :)